### PR TITLE
Private events

### PR DESCRIPTION
**Pull request by Tresor11**
This PR is aimed at implementing Authentication as well as Event creation with Rails has-many through associations. A user can see the event creator by calling the creator method saved on the Event's instance. A user can also view who is attending a specific event by calling the attendees method on the Event's instance.

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class Event < ApplicationRecord
+  belongs_to :creator, class_name: 'User'
+  has_many :bookings, foreign_key: :booked_event_id
+  has_many :attendees, through: :bookings, source: :event_attendee
+  scope :upcoming, -> { where('date > ?', Date.today) }
+  scope :past, -> { where('date < ?', Date.today) }
+  # Ex:- scope :active, -> {where(:active => true)}
```
Comment (ebukaume) [inline app/models/event.rb]: Committing commented code is a bad practice. Please remove

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        </ol>
+    </div>
+    <div class="col-4">
+        <h5>Past events:</h5>
+        <ol>
+          <%@past.each do |x|%>
+            <li><%= link_to  "#{x.title}", x%></li>
+          <%end%>
+        </ol>
+    </div>
```
Comment (ebukaume) [inline app/views/events/index.html.erb]: [OPTIONAL]
Consider having a partial for the list items to DRY out your code

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    </div>
+    <div class="col-4">
+        <h5>Previously attended events:</h5>
+        <ul>
+          <%@past.each do |x|%>
+            <li><%=x.title%></li>
+          <%end%>
+        </ul>
+    </div>
+</div>
```
Comment (ebukaume) [inline app/views/users/show.html.erb:32]: [OPTIONAL]
Here could also use some DRYing out. Partials to the rescue.

**Review by ebukaume — CHANGES_REQUESTED** [review]
## Changes Required! :stop_sign: 

Hi @Tresor11  and @mrnadaara :wave:

You guys are doing great. Let's address these issues.

* [ ] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [ ] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [ ] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

**Fix these issues and others below, then submit another code review request**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Review by Redvanisation — APPROVED** [review]
# Project approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀

**Comment by Tresor11** [comment]
Thank you we are going to use partials to make the code more DRY :smiley:

**Pull request merged**